app/services/bulk_upload/lettings/year2024/row_parser.rb

```diff
@@ -0,0 +1,1468 @@
+  end
+
+  def validate_managing_org_data_given
+    if field_2.blank?
+      block_log_creation!
+      errors.add(:field_2, "The managing organisation code is incorrect", category: :setup)
+    end
+  end
+
+  def validate_owning_org_owns_stock
+    if owning_organisation && !owning_organisation.holds_own_stock?
+      block_log_creation!
+
+      if errors[:field_1].blank?
+        errors.add(:field_1, "The owning organisation code provided is for an organisation that does not own stock", category: :setup)
+      end
+    end
+  end
+
+  def validate_owning_org_exists
+    if owning_organisation.nil?
+      block_log_creation!
+
+      if errors[:field_1].blank?
+        errors.add(:field_1, "The owning organisation code is incorrect", category: :setup)
+      end
+    end
+  end
+
+  def validate_owning_org_data_given
+    if field_1.blank?
+      block_log_creation!
+      errors.add(:field_1, I18n.t("validations.not_answered", question: "owning organisation"), category: :setup)
+    end
+  end
+
+  def validate_owning_org_permitted
+    if owning_organisation && !bulk_upload.user.organisation.affiliated_stock_owners.include?(owning_organisation)
+      block_log_creation!
+
+      if errors[:field_1].blank?
+        errors.add(:field_1, "You do not have permission to add logs for this owning organisation", category: :setup)
+      end
+    end
+  end
+
+  def setup_question?(question)
+    log.form.setup_sections[0].subsections[0].questions.include?(question)
+  end
+
+  def validate_if_log_already_exists
+    if log_already_exists?
+      error_message = "This is a duplicate log"
+
+      errors.add(:field_1, error_message) # owning_organisation
+      errors.add(:field_8, error_message) # startdate
+      errors.add(:field_9, error_message) # startdate
+      errors.add(:field_10, error_message) # startdate
+      errors.add(:field_13, error_message) # tenancycode
+      errors.add(:field_6, error_message) if !general_needs? && :field_6.present? # location
+      errors.add(:field_5, error_message) if !general_needs? && :field_6.blank? # add to Scheme field as unclear whether log uses New or Old CORE ids
+      errors.add(:field_21, error_message) unless supported_housing? # postcode_full
+      errors.add(:field_22, error_message) unless supported_housing? # postcode_full
+      errors.add(:field_23, error_message) unless supported_housing? # la
+      errors.add(:field_42, error_message) # age1
+      errors.add(:field_43, error_message) # sex1
+      errors.add(:field_46, error_message) # ecstat1
+      errors.add(:field_124, error_message) if log.chcharge.present? # chcharge
+      errors.add(:field_122, error_message) unless general_needs? # household_charge
+      errors.add(:field_125, error_message) # brent
+      errors.add(:field_126, error_message) # scharge
+      errors.add(:field_127, error_message) # pscharge
+      errors.add(:field_128, error_message) # chcharge
+    end
+  end
+
+  def field_mapping_for_errors
+    {
```

Comment:
why have we removed tcharge and offered from this?

Reply:
We're no longer asking offered question and tcharge I'll add back and will map to the other charges fields (we're no longer ask tcharge either)